subsys/bluetooth/host/classic/avrcp.c

```diff
@@ -0,0 +1,467 @@
+/** @file
+ *  @brief Audio Video Remote Control Profile
+ */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ * Copyright (C) 2024 Xiaomi Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <errno.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/printk.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/classic/avrcp.h>
+#include <zephyr/bluetooth/classic/sdp.h>
+#include <zephyr/bluetooth/l2cap.h>
+
+#include "host/hci_core.h"
+#include "host/conn_internal.h"
+#include "host/l2cap_internal.h"
+#include "avctp_internal.h"
+#include "avrcp_internal.h"
+
+#define LOG_LEVEL CONFIG_BT_AVRCP_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_avrcp);
+
+struct bt_avrcp {
+	struct bt_avctp session;
+	struct bt_avrcp_req req;
+	struct k_work_delayable timeout_work;
+	uint8_t local_tid;
+};
+
+static struct bt_avrcp_cb *avrcp_cb;
+
+#define AVRCP_TIMEOUT K_SECONDS(3) /* Shell be greater than TMTP (1000ms) */
+#define AVRCP_AVCTP(_avctp) CONTAINER_OF(_avctp, struct bt_avrcp, session)
+#define AVRCP_KWORK(_work) CONTAINER_OF(CONTAINER_OF(_work, struct k_work_delayable, work),\
+										struct bt_avrcp, timeout_work)
```

Comment:
too long line and also not correctly aligned

Reply:
Done.